### ISSUE TITLE
The ap of training FasterRCNN on VOC0712 is slow by using one 1080TI.

### ISSUE THREAD

**Issue by xtanitfy** [tensorpack/tensorpack#1023]
(1) **If you're using examples, what's the command you run:**
           ./train.py --config \
    MODE_MASK=False MODE_FPN=True \
    DATA.BASEDIR=DataSet/COCO_VOC \
    BACKBONE.WEIGHTS=pretrain_model/ImageNet-R50-AlignPadding.npz 
(2) **If you're using examples, have you made any changes to the examples? Paste them here:**
      I convert the voc to COCO format.

(3) My environment:
      tf:                 1.9.0
     hardware:     only one 1080TI
     dataset:        voc0712 of coco format 
（4）The part of my train log as such:
   #################################################################################################|500/500[01:51<00:00, 4.49it/s]
[1220 19:43:53 @base.py:282] Epoch 135 (global_step 67500) finished, time:1 minute 51 seconds.
[1220 19:43:53 @graph.py:73] Running Op sync_variables/sync_variables_from_main_tower ...
[1220 19:43:53 @misc.py:109] Estimated Time Left: 7 days 6 hours 35 minutes 32 seconds
[1220 19:43:54 @monitor.py:459] GPUUtil/0: 74.73
[1220 19:43:54 @monitor.py:459] PeakMemory(MB)/gpu:0: 6902.3
[1220 19:43:54 @monitor.py:459] QueueInput/queue_size: 50
[1220 19:43:54 @monitor.py:459] fastrcnn_losses/box_loss: 0.080622
[1220 19:43:54 @monitor.py:459] fastrcnn_losses/label_loss: 0.13035
[1220 19:43:54 @monitor.py:459] fastrcnn_losses/label_metrics/accuracy: 0.95448
[1220 19:43:54 @monitor.py:459] fastrcnn_losses/label_metrics/false_negative: 0.2658
[1220 19:43:54 @monitor.py:459] fastrcnn_losses/label_metrics/fg_accuracy: 0.66653
[1220 19:43:54 @monitor.py:459] fastrcnn_losses/num_fg_label: 42.771
[1220 19:43:54 @monitor.py:459] learning_rate: 0.01
[1220 19:43:54 @monitor.py:459] multilevel_roi_align/fpn_map_rois_to_levels/num_roi_level2: 261.75
[1220 19:43:54 @monitor.py:459] multilevel_roi_align/fpn_map_rois_to_levels/num_roi_level3: 119.58
[1220 19:43:54 @monitor.py:459] multilevel_roi_align/fpn_map_rois_to_levels/num_roi_level4: 106.42
[1220 19:43:54 @monitor.py:459] multilevel_roi_align/fpn_map_rois_to_levels/num_roi_level5: 24.261
[1220 19:43:54 @monitor.py:459] rpn_losses/box_loss: 0.019923
[1220 19:43:54 @monitor.py:459] rpn_losses/label_loss: 0.017205
[1220 19:43:54 @monitor.py:459] rpn_losses/level2/box_loss: 0.0021779
[1220 19:43:54 @monitor.py:459] rpn_losses/level2/label_loss: 0.0052635
[1220 19:43:54 @monitor.py:459] rpn_losses/level2/label_metrics/precision_th0.1: 0.42625
[1220 19:43:54 @monitor.py:459] rpn_losses/level2/label_metrics/precision_th0.2: 0.49367
[1220 19:43:54 @monitor.py:459] rpn_losses/level2/label_metrics/precision_th0.5: 0.5314
[1220 19:43:54 @monitor.py:459] rpn_losses/level2/label_metrics/recall_th0.1: 0.53351
[1220 19:43:54 @monitor.py:459] rpn_losses/level2/label_metrics/recall_th0.2: 0.53206
[1220 19:43:54 @monitor.py:459] rpn_losses/level2/label_metrics/recall_th0.5: 0.52253
[1220 19:43:54 @monitor.py:459] rpn_losses/level2/num_pos_anchor: 2.6011
[1220 19:43:54 @monitor.py:459] rpn_losses/level2/num_valid_anchor: 188.56
[1220 19:43:54 @monitor.py:459] rpn_losses/level3/box_loss: 0.0033833
[1220 19:43:54 @monitor.py:459] rpn_losses/level3/label_loss: 0.0032648
[1220 19:43:54 @monitor.py:459] rpn_losses/level3/label_metrics/precision_th0.1: 0.50834
[1220 19:43:54 @monitor.py:459] rpn_losses/level3/label_metrics/precision_th0.2: 0.53245
[1220 19:43:54 @monitor.py:459] rpn_losses/level3/label_metrics/precision_th0.5: 0.54092
[1220 19:43:54 @monitor.py:459] rpn_losses/level3/label_metrics/recall_th0.1: 0.58832
[1220 19:43:54 @monitor.py:459] rpn_losses/level3/label_metrics/recall_th0.2: 0.58829
[1220 19:43:54 @monitor.py:459] rpn_losses/level3/label_metrics/recall_th0.5: 0.58055
[1220 19:43:54 @monitor.py:459] rpn_losses/level3/num_pos_anchor: 3.7471
[1220 19:43:54 @monitor.py:459] rpn_losses/level3/num_valid_anchor: 46.942
[1220 19:43:54 @monitor.py:459] rpn_losses/level4/box_loss: 0.00079764
[1220 19:43:54 @monitor.py:459] rpn_losses/level4/label_loss: 0.0019006
[1220 19:43:54 @monitor.py:459] rpn_losses/level4/label_metrics/precision_th0.1: 0.39629
[1220 19:43:54 @monitor.py:459] rpn_losses/level4/label_metrics/precision_th0.2: 0.45111
[1220 19:43:54 @monitor.py:459] rpn_losses/level4/label_metrics/precision_th0.5: 0.56285
[1220 19:43:54 @monitor.py:459] rpn_losses/level4/label_metrics/recall_th0.1: 0.59904
[1220 19:43:54 @monitor.py:459] rpn_losses/level4/label_metrics/recall_th0.2: 0.5986
[1220 19:43:54 @monitor.py:459] rpn_losses/level4/label_metrics/recall_th0.5: 0.57843
[1220 19:43:54 @monitor.py:459] rpn_losses/level4/num_pos_anchor: 1.0174
[1220 19:43:54 @monitor.py:459] rpn_losses/level4/num_valid_anchor: 9.8574
[1220 19:43:54 @monitor.py:459] rpn_losses/level5/box_loss: 0.0030008
[1220 19:43:54 @monitor.py:459] rpn_losses/level5/label_loss: 0.0048236
[1220 19:43:54 @monitor.py:459] rpn_losses/level5/label_metrics/precision_th0.1: 0.50047
[1220 19:43:54 @monitor.py:459] rpn_losses/level5/label_metrics/precision_th0.2: 0.52787
[1220 19:43:54 @monitor.py:459] rpn_losses/level5/label_metrics/precision_th0.5: 0.57011
[1220 19:43:54 @monitor.py:459] rpn_losses/level5/label_metrics/recall_th0.1: 0.69719
[1220 19:43:54 @monitor.py:459] rpn_losses/level5/label_metrics/recall_th0.2: 0.69719
[1220 19:43:54 @monitor.py:459] rpn_losses/level5/label_metrics/recall_th0.5: 0.68959
[1220 19:43:54 @monitor.py:459] rpn_losses/level5/num_pos_anchor: 2.0413
[1220 19:43:54 @monitor.py:459] rpn_losses/level5/num_valid_anchor: 3.2903
[1220 19:43:54 @monitor.py:459] rpn_losses/level6/box_loss: 0.010564
[1220 19:43:54 @monitor.py:459] rpn_losses/level6/label_loss: 0.0019524
[1220 19:43:54 @monitor.py:459] rpn_losses/level6/label_metrics/precision_th0.1: 0.87381
[1220 19:43:54 @monitor.py:459] rpn_losses/level6/label_metrics/precision_th0.2: 0.87381
[1220 19:43:54 @monitor.py:459] rpn_losses/level6/label_metrics/precision_th0.5: 0.87675
[1220 19:43:54 @monitor.py:459] rpn_losses/level6/label_metrics/recall_th0.1: 0.87891
[1220 19:43:54 @monitor.py:459] rpn_losses/level6/label_metrics/recall_th0.2: 0.87891
[1220 19:43:54 @monitor.py:459] rpn_losses/level6/label_metrics/recall_th0.5: 0.87538
[1220 19:43:54 @monitor.py:459] rpn_losses/level6/num_pos_anchor: 7.3006
[1220 19:43:54 @monitor.py:459] rpn_losses/level6/num_valid_anchor: 7.3481
[1220 19:43:54 @monitor.py:459] sample_fast_rcnn_targets/num_bg: 469.23
[1220 19:43:54 @monitor.py:459] sample_fast_rcnn_targets/num_fg: 42.771
[1220 19:43:54 @monitor.py:459] sample_fast_rcnn_targets/proposal_metrics/best_iou_per_gt: 0.77643
[1220 19:43:54 @monitor.py:459] sample_fast_rcnn_targets/proposal_metrics/recall_iou0.3: 0.99868
[1220 19:43:54 @monitor.py:459] sample_fast_rcnn_targets/proposal_metrics/recall_iou0.5: 0.99867
[1220 19:43:54 @monitor.py:459] total_cost: 0.62056
[1220 19:43:54 @monitor.py:459] wd_cost: 0.37246
100%|######################################################################################################|500/500[01:52<00:00, 4.44it/s]
[1221 09:21:06 @base.py:282] Epoch 325 (global_step 162500) finished, time:1 minute 52 seconds.
[1221 09:21:06 @graph.py:73] Running Op sync_variables/sync_variables_from_main_tower ...
[1221 09:21:06 @train.py:504] Running evaluation ...
100%|█████████████████████████████████████████████████████████████████████████████████████████████████| 4952/4952 [07:21<00:00, 11.37it/s]
loading annotations into memory...
Done (t=0.12s)
creating index...
index created!
Loading and preparing results...
DONE (t=0.65s)
creating index...
index created!
Running per image evaluation...
Evaluate annotation type *bbox*
DONE (t=11.24s).
Accumulating evaluation results...
DONE (t=3.92s).
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.038
 Average Precision  (AP) @[ IoU=0.50      | area=   all | maxDets=100 ] = 0.075
 Average Precision  (AP) @[ IoU=0.75      | area=   all | maxDets=100 ] = 0.033
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.054
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.047
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.041
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=  1 ] = 0.035
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets= 10 ] = 0.061
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.063
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.063
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.063
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.063
  
(5) My question：
 Why the ap is too slow after so many steps?
 Looking forward to your reply.Thanks!

**Comment by xtanitfy** [comment]
I found that your version of training is very fast compared to the version which link is 
 https://github.com/matterport/Mask_RCNN and can be flexibly configured.
I guess the reason for the slow convergence is because the number of graphics card is too small and the batch is very small.

**Comment by ppwwyyxx** [comment]
See https://github.com/tensorpack/tensorpack/blob/master/.github/ISSUE_TEMPLATE.md:

> "The examples do not perform as expected after I change the models/dataset/parameters/etc." Tensorpack maintainers make sure the examples perform well without modifications. But it's your job to pick the model and parameters that are suitable for your own situation. We cannot help with such questions unless they appear to be a bug in tensorpack.
